zerver/test_messages.py

```diff
@@ -1432,6 +1434,21 @@ def test_basics(self):
         self.assertTrue(Message.content_has_link('https://humbug-user-uploads.s3.amazonaws.com/sX_TIQx/screen-shot.jpg'))
         self.assertTrue(Message.content_has_link('https://humbug-user-uploads-test.s3.amazonaws.com/sX_TIQx/screen-shot.jpg'))
 
+    def test_claim_attachment(self):
+        #create dummy DB entry
+        sender_email = "hamlet@zulip.com"
+        user_profile = get_user_profile_by_email(sender_email)
+        path_id = '1/31/4CBjtTLYZhk66pZrF8hnYGwc/zulip.txt'
+        store_as_temp('zulip.txt', path_id, user_profile)
+
+        #send message referring the attachment
+        self.subscribe_to_stream(sender_email, "Denmark")
+        body = '[zulip.txt](http://localhost:9991/user_uploads/1/31/4CBjtTLYZhk66pZrF8hnYGwc/zulip.txt)'
+        self.send_message(sender_email, "Denmark", Recipient.STREAM, body, "test")
+
+        attachment = Attachment.objects.get(path_id=path_id)
+        self.assertTrue(attachment.is_claimed())
```

Comment:
Let's add another similar test that just sends something of the form "http://localhost:9991/user_uploads/1/31/4CBjtTLYZhk66pZrF8hnYGwc/zulip.txt" (no `[]()` around it)